### PR TITLE
Add AutoEmote comp/system, updates to zombie code

### PR DESCRIPTION
**Pull request by 0x6273**
This adds component and system for making an entity automatically perform emotes at random intervals.

The component stores a set of prototype IDs for the emotes, so multiple systems can use it simultaneously with different emotes. For example, you could have a zombified cluwne that both groans and honks randomly.

This PR also makes zombies use this for random groaning. Groaning when taking damage is still coupled to zombie code, but it was refactored slightly to not do time accumulators in the ZombieSystem Update method.

Related: #13850, #13367

### CODE REVIEW

File: Content.Server/Chat/AutoEmoteComponent.cs
Comment (metalgearsloth) [inline Content.Server/Chat/AutoEmoteComponent.cs:31]: This should be a datafield + timeoffsetserializer + updating on unpause.

File: Content.Server/Chat/AutoEmoteComponent.cs
Comment (metalgearsloth) [inline Content.Server/Chat/AutoEmoteComponent.cs:25]: Do we need multiple emotes?

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chat/Systems/AutoEmoteSystem.cs:63]: Init shouldn't implicitly reset these because the comp may be deserialized.

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chat/Systems/AutoEmoteSystem.cs:30]: You should put CurTime in a local variable because it gets calculated every time.

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chat/Systems/AutoEmoteSystem.cs:89]: Should this be logging on failures?

Also preformed

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Chat/AutoEmoteComponent.cs
Comment (0x6273) [reply Content.Server/Chat/AutoEmoteComponent.cs:25]: brainfood wants cluwnes to have multiple emotes like honks, giggles, so yes I think it's needed. Also if a Cluwne turns into a zombie it's possible for it to groan while still honking and giggling.

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (0x6273) [reply Content.Server/Chat/Systems/AutoEmoteSystem.cs:64]: So I want this to work not just with `AddEmote` method but also if you have the component on an entity prototype like 
```yaml
- type: AutoEmote
  emotes: ["ZombieGroan"]
```

So I need to start the timers the first time the entity is spawned, but not if it's deserialized from like a map/grid that's initialized? Do I use `MapInit` event for this instead?

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (0x6273) [reply Content.Server/Chat/Systems/AutoEmoteSystem.cs]: Not sure, I made it not error on missing comp now

File: Content.Server/Chat/AutoEmoteComponent.cs
Comment (0x6273) [reply Content.Server/Chat/AutoEmoteComponent.cs:31]: I think maybe only EmoteTimers needs to be datafield and this is just always derived from it on init? This should always point to the lowest TimeSpan in EmoteTimers.

Is there a component that handles pausing and unpausing timespans properly that I can look at?

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (brainfood1183) [reply Content.Server/Chat/Systems/AutoEmoteSystem.cs]: Hows the progress going on this? waiting for another review?

**Comment by 0x6273** [comment]
I think this is blocked by https://github.com/space-wizards/RobustToolbox/issues/3768 if resetting the timers on init is not acceptable.

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chat/Systems/AutoEmoteSystem.cs:31]: Should get the time outside of the loop.

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (metalgearsloth) [reply Content.Server/Chat/Systems/AutoEmoteSystem.cs:66]: Yeah MapInit would be preferable I think.

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chat/Systems/AutoEmoteSystem.cs:74]: This should just be += pausetime right?

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chat/Systems/AutoEmoteSystem.cs:79]:
```suggestion
    /// <summary>
    /// Try to add an emote to the entity, which will be performed at an interval.
    /// </summary>
```

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chat/Systems/AutoEmoteSystem.cs:88]: Should add a debug assert that the prototype exists

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chat/Systems/AutoEmoteSystem.cs:105]: Same here

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chat/Systems/AutoEmoteSystem.cs:113]: Just use the prototype directly as the update caller already has it and it saves having to do Index again.

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (metalgearsloth) [inline Content.Server/Chat/Systems/AutoEmoteSystem.cs:125]: Put CurTime in a local variable.

**Review by metalgearsloth — DISMISSED** [review]

File: Content.Server/Chat/Systems/AutoEmoteSystem.cs
Comment (0x6273) [reply Content.Server/Chat/Systems/AutoEmoteSystem.cs:89]: I don't think this is needed here since ResetTimer is gonna index it anyway. I just added the assert in RemoveEmote.

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**
